components/ModEditSubmissionTopbar.vue

```diff
@@ -76,4 +74,6 @@ const acceptSubmission = () => {
 const rejectSubmission = () => {
     console.log('You are rejected')
 }
+
+const submissionId = computed(() => route.params.id as string)
```

Comment:
This is looking good and happy to see some vue router action! I think there's a few things that we can adjust here so I'll just put it in one comment

## feature: adding support for refreshing page or coming from a shared url. 

You could do this is the PR or separate. I'd say add it here as it will require a few changes to the code to make it work. 

#### what needs to change 
- since we could view or edit submissions, you should specify that this is `/editsubmission/id` 
  - this will also help when you refresh to know which page you're going to 
- You need to make the pages based on the screen again as the submission id could be for more than one screen
- I would recommend setting back to `selectedsubmissionId` again as it's a little more clear which submission id you're referring to. (For example as opposed to `submissionIdToDelete`)
- I would probably set the selectedsubmissionId in the store as it's shared and part of the route
- you can set the screen based on the route now if you have the `editSubmission/` route. 
- You need to add in logic for when the mod dashboard is first loaded, it should check the route and go to the edit page immediately. This could be tricky to not have the page flash quickly. Let me know if you want to know how or you want to try and figure it out 😎

Reply:
"You need to make the pages based on the screen again as the submission id could be for more than one screen"


I do not understand this. When would the id be used for a different screen. It was conditional based off the selected submissionId before. We are going to add more screens? Wouldn't we use the pagination to continue to conditionally render everywhere? Is it not adding complexity to use router parameters and a store where they can do the same thing? Or does it just give way more control?

Since you suggested `editSubmission/submissionId` couldn't we use the router in place of the store? Is this bad practice? Less reliable?

Reply:
Let's huddle tomorrow  and we can talk through it 👌

Reply:
I think I understand now. If we have another variable we set in the store like `isSubmissionToDelete` we could show a different screen or a modal or something based on the screen. E.G. `ModerationScreen.ConfrimDelete`.

Reply:
> This is looking good and happy to see some vue router action! I think there's a few things that we can adjust here so I'll just put it in one comment
> 
> ## feature: adding support for refreshing page or coming from a shared url.
> You could do this is the PR or separate. I'd say add it here as it will require a few changes to the code to make it work.
> 
> #### what needs to change
> * since we could view or edit submissions, you should specify that this is `/editsubmission/id`
>   
>   * this will also help when you refresh to know which page you're going to
> * You need to make the pages based on the screen again as the submission id could be for more than one screen
> * I would recommend setting back to `selectedsubmissionId` again as it's a little more clear which submission id you're referring to. (For example as opposed to `submissionIdToDelete`)
> * I would probably set the selectedsubmissionId in the store as it's shared and part of the route
> * you can set the screen based on the route now if you have the `editSubmission/` route.
> * You need to add in logic for when the mod dashboard is first loaded, it should check the route and go to the edit page immediately. This could be tricky to not have the page flash quickly. Let me know if you want to know how or you want to try and figure it out 😎

Updated based on feedback plus reasoning.
-The path now is `/editsubmission/id` for the form.
- The pages are reverted to the screen again and my understanding is show as above.
- The selectedSubmissionId is set in the store if we need to use it other places and watched in order to update everywhere.
-I added a ref that is a boolean to check the path. and added different conditions to show the different page so the one that shows the link to the form isn't shown unless auth fails. Otherwise the user is shown 'Checking Authentication'. I left that in for all the paths cuz it is happening and I want the user to see that it is still a safe and protected path.